include/El/core/Grid.hpp

```diff
@@ -132,6 +134,92 @@ inline void AssertSameGrids( const Grid& g1, const Grid& g2, Args&... args )
     AssertSameGrids( g2, args... );
 }
 
+inline bool GridCompare
```

Comment:
In that case, if it isn't possible to simply reuse ``g1``, then I would recommend having ``GridCompare`` check if the output of ``MPI_Comm_compare`` and ``MPI_Group_compare`` are ``MPI_IDENT`` in addition to the height being the same. If the inputs to the constructor are the same, the outputs should be too (and this would eliminate the need for the ``friend`` declaration).